File: Guns/src/main/webapp/static/modular/p/pUserBaseInfo/pUserBaseInfo.js
/**
 * 管理初始化
 */
var PUserBaseInfo = {
    id: "PUserBaseInfoTable",	//表格id
    seItem: null,		//选中的条目
    table: null,
    layerIndex: -1
};

/**
 * 初始化表格的列
 */
PUserBaseInfo.initColumn = function () {
    return [
        {field: 'selectItem', radio: true},
            {title: '', field: 'id', visible: true, align: 'center', valign: 'middle'},
            {title: '用户编号', field: 'userSn', visible: true, align: 'center', valign: 'middle'},
            {title: '用户手机号', field: 'mobile', visible: true, align: 'center', valign: 'middle'},
            {title: '密码', field: 'password', visible: true, align: 'center', valign: 'middle'},
            {title: 'md5密码盐（考虑到APP端密码加密规则的兼容性，此值暂不使用）', field: 'salt', visible: true, align: 'center', valign: 'middle'},
            {title: '', field: 'nickname', visible: true, align: 'center', valign: 'middle'},
            {title: '头像', field: 'headThumb', visible: true, align: 'center', valign: 'middle'},
            {title: '位置数据', field: 'locationArea', visible: true, align: 'center', valign: 'middle'},
            {title: '地区', field: 'region', visible: true, align: 'center', valign: 'middle'},
            {title: '个人介绍', field: 'synopsis', visible: true, align: 'center', valign: 'middle'},
            {title: '车牌号', field: 'licensePlateNumber', visible: true, align: 'center', valign: 'middle'},
            {title: '状态：1：正常使用，2：等在认证: 3：被冻结', field: 'starts', visible: true, align: 'center', valign: 'middle'},
            {title: '身份证号码', field: 'identityCard', visible: true, align: 'center', valign: 'middle'},
            {title: '信用分', field: 'integral', visible: true, align: 'center', valign: 'middle'},
            {title: '用户身份，1：乘客；2：司机', field: 'identity', visible: true, align: 'center', valign: 'middle'},
            {title: '微信openid', field: 'openid', visible: true, align: 'center', valign: 'middle'},
            {title: '微信授权用户唯一标识', field: 'unionid', visible: true, align: 'center', valign: 'middle'},
            {title: 'app的openid', field: 'appOpenId', visible: true, align: 'center', valign: 'middle'},
            {title: '融云token', field: 'token', visible: true, align: 'center', valign: 'middle'},
            {title: '姓名', field: 'name', visible: true, align: 'center', valign: 'middle'},
            {title: '性别，1男，2女，保密', field: 'sex', visible: true, align: 'center', valign: 'middle'},
            {title: '用户佣金余额', field: 'money', visible: true, align: 'center', valign: 'middle'},
            {title: '我的邀请码', field: 'myInvite', visible: true, align: 'center', valign: 'middle'},
            {title: '推荐人', field: 'pid', visible: true, align: 'center', valign: 'middle'},
            {title: '层级数，0,1,2,3,4....', field: 'level', visible: true, align: 'center', valign: 'middle'},
            {title: '排序', field: 'sortNum', visible: true, align: 'center', valign: 'middle'},
            {title: '备注', field: 'remark', visible: true, align: 'center', valign: 'middle'},
            {title: '是否启用,0不启用，1启用', field: 'isEnable', visible: true, align: 'center', valign: 'middle'},
            {title: '创建人', field: 'creator', visible: true, align: 'center', valign: 'middle'},
            {title: '修改人', field: 'modifier', visible: true, align: 'center', valign: 'middle'},
            {title: '创建时间', field: 'created', visible: true, align: 'center', valign: 'middle'},
            {title: '最后修改时间', field: 'modified', visible: true, align: 'center', valign: 'middle'},
            {title: '0正常，1删除', field: 'isDeleted', visible: true, align: 'center', valign: 'middle'}
    ];
};

/**
 * 检查是否选中
 */
PUserBaseInfo.check = function () {
    var selected = $('#' + this.id).bootstrapTable('getSelections');
    if(selected.length == 0){
        Feng.info("请先选中表格中的某一记录！");
        return false;
    }else{
        PUserBaseInfo.seItem = selected[0];
        return true;
    }
};

/**
 * 点击添加
 */
PUserBaseInfo.openAddPUserBaseInfo = function () {
    var index = layer.open({
        type: 2,
        title: '添加',
        area: ['800px', '420px'], //宽高
        fix: false, //不固定
        maxmin: true,
        content: Feng.ctxPath + '/pUserBaseInfo/pUserBaseInfo_add'
    });
    this.layerIndex = index;
};

/**
 * 打开查看详情
 */
PUserBaseInfo.openPUserBaseInfoDetail = function () {
    if (this.check()) {
        var index = layer.open({
            type: 2,
            title: '详情',
            area: ['800px', '420px'], //宽高
            fix: false, //不固定
            maxmin: true,
            content: Feng.ctxPath + '/pUserBaseInfo/pUserBaseInfo_update/' + PUserBaseInfo.seItem.id
        });
        this.layerIndex = index;
    }
};

/**
 * 删除
 */
PUserBaseInfo.delete = function () {
    if (this.check()) {
        var ajax = new $ax(Feng.ctxPath + "/pUserBaseInfo/delete", function (data) {
            Feng.success("删除成功!");
            PUserBaseInfo.table.refresh();
        }, function (data) {
            Feng.error("删除失败!" + data.responseJSON.message + "!");
        });
        ajax.set("pUserBaseInfoId",this.seItem.id);
        ajax.start();
    }
};

/**
 * 查询列表
 */
PUserBaseInfo.search = function () {
    var queryData = {};
    queryData['condition'] = $("#condition").val();
    PUserBaseInfo.table.refresh({query: queryData});
};

$(function () {
    var defaultColunms = PUserBaseInfo.initColumn();
    var table = new BSTable(PUserBaseInfo.id, "/pUserBaseInfo/list", defaultColunms);
    table.setPaginationType("client");
    PUserBaseInfo.table = table.init();
});

File: Guns/src/main/java/cn/stylefeng/guns/modular/p/model/PUserBaseInfo.java
package cn.stylefeng.guns.modular.p.model;

import com.baomidou.mybatisplus.enums.IdType;
import java.math.BigDecimal;
import java.util.Date;
import com.baomidou.mybatisplus.annotations.TableId;
import com.baomidou.mybatisplus.annotations.TableField;
import com.baomidou.mybatisplus.activerecord.Model;
import com.baomidou.mybatisplus.annotations.TableName;
import java.io.Serializable;

/**
 * <p>
 * 乘客和司机表
 * </p>
 *
 * @author shiye
 * @since 2019-04-03
 */
@TableName("p_user_base_info")
public class PUserBaseInfo extends Model<PUserBaseInfo> {

    private static final long serialVersionUID = 1L;

    @TableId(value = "id", type = IdType.AUTO)
    private Integer id;
    /**
     * 用户编号
     */
    @TableField("user_sn")
    private String userSn;
    /**
     * 用户手机号
     */
    private String mobile;
    /**
     * 密码
     */
    private String password;
    /**
     * md5密码盐（考虑到APP端密码加密规则的兼容性，此值暂不使用）
     */
    private String salt;
    private String nickname;
    /**
     * 头像
     */
    @TableField("head_thumb")
    private String headThumb;
    /**
     * 位置数据
     */
    @TableField("location_area")
    private String locationArea;
    /**
     * 地区
     */
    private String region;
    /**
     * 个人介绍
     */
    private String synopsis;
    /**
     * 车牌号
     */
    @TableField("license_plate_number")
    private String licensePlateNumber;
    /**
     * 状态：1：正常使用，2：等在认证: 3：被冻结
     */
    private Integer starts;
    /**
     * 身份证号码
     */
    @TableField("identity_card")
    private String identityCard;
    /**
     * 信用分
     */
    private BigDecimal integral;
    /**
     * 用户身份，1：乘客；2：司机
     */
    private Integer identity;
    /**
     * 微信openid
     */
    private String openid;
    /**
     * 微信授权用户唯一标识
     */
    private String unionid;
    /**
     * app的openid
     */
    @TableField("app_open_id")
    private String appOpenId;
    /**
     * 融云token
     */
    private String token;
    /**
     * 姓名
     */
    private String name;
    /**
     * 性别，1男，2女，保密
     */
    private String sex;
    /**
     * 用户佣金余额
     */
    private BigDecimal money;
    /**
     * 我的邀请码
     */
    @TableField("my_invite")
    private String myInvite;
    /**
     * 推荐人
     */
    private Integer pid;
    /**
     * 层级数，0,1,2,3,4....
     */
    private Integer level;
    /**
     * 排序
     */
    @TableField("sort_num")
    private Integer sortNum;
    /**
     * 备注
     */
    private String remark;
    /**
     * 是否启用,0不启用，1启用
     */
    @TableField("is_enable")
    private Integer isEnable;
    /**
     * 创建人
     */
    private Integer creator;
    /**
     * 修改人
     */
    private Integer modifier;
    /**
     * 创建时间
     */
    private Date created;
    /**
     * 最后修改时间
     */
    private Date modified;
    /**
     * 0正常，1删除
     */
    @TableField("is_deleted")
    private Integer isDeleted;


    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getUserSn() {
        return userSn;
    }

    public void setUserSn(String userSn) {
        this.userSn = userSn;
    }

    public String getMobile() {
        return mobile;
    }

    public void setMobile(String mobile) {
        this.mobile = mobile;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getSalt() {
        return salt;
    }

    public void setSalt(String salt) {
        this.salt = salt;
    }

    public String getNickname() {
        return nickname;
    }

    public void setNickname(String nickname) {
        this.nickname = nickname;
    }

    public String getHeadThumb() {
        return headThumb;
    }

    public void setHeadThumb(String headThumb) {
        this.headThumb = headThumb;
    }

    public String getLocationArea() {
        return locationArea;
    }

    public void setLocationArea(String locationArea) {
        this.locationArea = locationArea;
    }

    public String getRegion() {
        return region;
    }

    public void setRegion(String region) {
        this.region = region;
    }

    public String getSynopsis() {
        return synopsis;
    }

    public void setSynopsis(String synopsis) {
        this.synopsis = synopsis;
    }

    public String getLicensePlateNumber() {
        return licensePlateNumber;
    }

    public void setLicensePlateNumber(String licensePlateNumber) {
        this.licensePlateNumber = licensePlateNumber;
    }

    public Integer getStarts() {
        return starts;
    }

    public void setStarts(Integer starts) {
        this.starts = starts;
    }

    public String getIdentityCard() {
        return identityCard;
    }

    public void setIdentityCard(String identityCard) {
        this.identityCard = identityCard;
    }

    public BigDecimal getIntegral() {
        return integral;
    }

    public void setIntegral(BigDecimal integral) {
        this.integral = integral;
    }

    public Integer getIdentity() {
        return identity;
    }

    public void setIdentity(Integer identity) {
        this.identity = identity;
    }

    public String getOpenid() {
        return openid;
    }

    public void setOpenid(String openid) {
        this.openid = openid;
    }

    public String getUnionid() {
        return unionid;
    }

    public void setUnionid(String unionid) {
        this.unionid = unionid;
    }

    public String getAppOpenId() {
        return appOpenId;
    }

    public void setAppOpenId(String appOpenId) {
        this.appOpenId = appOpenId;
    }

    public String getToken() {
        return token;
    }

    public void setToken(String token) {
        this.token = token;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getSex() {
        return sex;
    }

    public void setSex(String sex) {
        this.sex = sex;
    }

    public BigDecimal getMoney() {
        return money;
    }

    public void setMoney(BigDecimal money) {
        this.money = money;
    }

    public String getMyInvite() {
        return myInvite;
    }

    public void setMyInvite(String myInvite) {
        this.myInvite = myInvite;
    }

    public Integer getPid() {
        return pid;
    }

    public void setPid(Integer pid) {
        this.pid = pid;
    }

    public Integer getLevel() {
        return level;
    }

    public void setLevel(Integer level) {
        this.level = level;
    }

    public Integer getSortNum() {
        return sortNum;
    }

    public void setSortNum(Integer sortNum) {
        this.sortNum = sortNum;
    }

    public String getRemark() {
        return remark;
    }

    public void setRemark(String remark) {
        this.remark = remark;
    }

    public Integer getIsEnable() {
        return isEnable;
    }

    public void setIsEnable(Integer isEnable) {
        this.isEnable = isEnable;
    }

    public Integer getCreator() {
        return creator;
    }

    public void setCreator(Integer creator) {
        this.creator = creator;
    }

    public Integer getModifier() {
        return modifier;
    }

    public void setModifier(Integer modifier) {
        this.modifier = modifier;
    }

    public Date getCreated() {
        return created;
    }

    public void setCreated(Date created) {
        this.created = created;
    }

    public Date getModified() {
        return modified;
    }

    public void setModified(Date modified) {
        this.modified = modified;
    }

    public Integer getIsDeleted() {
        return isDeleted;
    }

    public void setIsDeleted(Integer isDeleted) {
        this.isDeleted = isDeleted;
    }

    @Override
    protected Serializable pkVal() {
        return this.id;
    }

    @Override
    public String toString() {
        return "PUserBaseInfo{" +
        ", id=" + id +
        ", userSn=" + userSn +
        ", mobile=" + mobile +
        ", password=" + password +
        ", salt=" + salt +
        ", nickname=" + nickname +
        ", headThumb=" + headThumb +
        ", locationArea=" + locationArea +
        ", region=" + region +
        ", synopsis=" + synopsis +
        ", licensePlateNumber=" + licensePlateNumber +
        ", starts=" + starts +
        ", identityCard=" + identityCard +
        ", integral=" + integral +
        ", identity=" + identity +
        ", openid=" + openid +
        ", unionid=" + unionid +
        ", appOpenId=" + appOpenId +
        ", token=" + token +
        ", name=" + name +
        ", sex=" + sex +
        ", money=" + money +
        ", myInvite=" + myInvite +
        ", pid=" + pid +
        ", level=" + level +
        ", sortNum=" + sortNum +
        ", remark=" + remark +
        ", isEnable=" + isEnable +
        ", creator=" + creator +
        ", modifier=" + modifier +
        ", created=" + created +
        ", modified=" + modified +
        ", isDeleted=" + isDeleted +
        "}";
    }
}
